search-processors/src/main/java/org/opensearch/searchpipelines/questionanswering/generative/prompt/PromptUtil.java

```diff
@@ -18,57 +18,144 @@
 package org.opensearch.searchpipelines.questionanswering.generative.prompt;
 
 import com.google.common.annotations.VisibleForTesting;
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonPrimitive;
 import lombok.AccessLevel;
+import lombok.Getter;
 import lombok.NoArgsConstructor;
 import org.apache.commons.text.StringEscapeUtils;
+import org.opensearch.ml.common.conversation.Interaction;
 
+import java.util.ArrayList;
 import java.util.List;
-import java.util.Locale;
 
 /**
+ * A utility class for producing prompts for LLMs.
+ *
  * TODO Should prompt engineering llm-specific?
  *
  */
 @NoArgsConstructor(access = AccessLevel.PRIVATE)
 public class PromptUtil {
 
+    public static final String DEFAULT_CHAT_COMPLETION_PROMPT_TEMPLATE =
+        "Generate a concise and informative answer in less than 100 words for the given question, taking into context: "
+            + "- An enumerated list of search results"
+            + "- A rephrase of the question that was used to generate the search results"
+            + "- The conversation history"
+            + "Cite search results using [${number}] notation."
+            + "Do not repeat yourself, and NEVER repeat anything in the chat history."
+            + "If there are any necessary steps or procedures in your answer, enumerate them.";
+
     private static final String roleUser = "user";
 
-    public static String getQuestionRephrasingPrompt(String originalQuestion, List<String> chatHistory) {
+    public static String getQuestionRephrasingPrompt(String originalQuestion, List<Interaction> chatHistory) {
         return null;
     }
 
-    public static String getChatCompletionPrompt(String question, List<String> chatHistory, List<String> contexts) {
+    public static String getChatCompletionPrompt(String question, List<Interaction> chatHistory, List<String> contexts) {
         return buildMessageParameter(question, chatHistory, contexts);
     }
 
+    enum Role {
```

Comment:
Can we rename Role to like ChatRole? I think it is specific for chat. The name "Role" is too general which could be confusing to the security role.

Reply:
@austintlee @HenryL27 do we have plan to address this?

Reply:
Sure.

Reply:
Just to clear, this enum is internal to PromptUtil, but I will rename it to ChatRole just to make it clearer.